Sources/Noora/Utilities/ProgressBar.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+protocol ProgressBar {
+    func startProgress(total: Int, interval: TimeInterval?, block: @escaping (String, Int) -> Void)
+    func stop()
+}
+final class DefaultProgressBar: ProgressBar {
```

Comment:
I'd merge all of this into the `ProgressBar` component to minimise the number of files making the codebase easier to navigate. If any logic from the component is reusable in the future, we can then make the decision to extract it.